pkg/services/serviceaccounts/manager/service.go

```diff
@@ -289,3 +307,55 @@ func validAPIKeyID(apiKeyID int64) error {
 	}
 	return nil
 }
+
+// hideApiKeysTabIfNoAPIKeysPresent is used to remove the apikeys tab from the admin tabs
+// if there are no apikeys, this makes it so that we only issue service account tokens instead
+func hideApiKeysTabIfNoAPIKeysPresent(l *log.ConcreteLogger, store *sqlstore.SQLStore, serviceAccountsStore *database.ServiceAccountsStoreImpl) {
+	var err error
+	ctx := context.Background()
+	sess := store.GetSqlxSession()
+	orgs := make([]*org.OrgDTO, 0)
+	err = sess.Select(ctx, &orgs, "SELECT id, name FROM org")
+	if err != nil {
+		l.Error(fmt.Sprintf("unable to query for org for hiding api keys: %e", err))
+		return
+	}
+	if len(orgs) != 1 {
+		l.Info("do not want to hide apikeys for more than 1 org")
+		return
+	}
+
+	orgID := orgs[0].ID
+
+	type result struct {
+		Count int64
+	}
+	r := result{}
+	// counting both service account tokens and api keys
+	err = sess.Get(ctx, &r, `SELECT COUNT(*) AS count FROM api_key WHERE org_id = ?`, orgID)
+	if err != nil {
+		l.Error(fmt.Sprintf("could not query for apikeys in org: %e", err))
+		return
+	}
+	if r.Count != 0 {
+		l.Debug("do not want to hide apikeys tab for org with apikeys")
+		return
+	}
+	// r.Count == 0
+	// we have no apikeys or service account tokens
+
+	// setting hide the apikeys tab for the org
+	err = serviceAccountsStore.HideApiKeysTab(ctx, orgID)
+	if err != nil {
+		l.Error(fmt.Sprintf("could not hide apikeys tab: %e", err))
+		return
+	}
+	// setting a global variable to hide the apikeys tab for newly created orgs
+	// used with check global hide apikeys tab
+	err = serviceAccountsStore.HideApiKeysTab(ctx, GloballyHideAPIKeysTabOrgID)
```

Comment:
Looks like API keys tab for new orgs will only be hidden if there's one org without keys/tokens exist (new Grafana instance). Is it by design? If yes, then we can make it simpler and always hide API keys tab if global flag set (in `pkg/services/navtree/navtreeimpl/admin.go`). Setting flag on org creation looks like a solution for case when we need to hide keys in some orgs (just created) and show in others (existing).

Reply:
> Looks like API keys tab for new orgs will only be hidden if there's one org without keys/tokens exist (new Grafana instance). Is it by design?

yeas, this is the "safest" option to detect the minimum amount required but still catch most instances that are created.

Reply:
discussed and we will check the global org id that is set on a new instance and check that in the admin.go index file to hide the tab, instead of having in the org creation